dockerfiles/incr.sh

```diff
@@ -80,9 +85,24 @@ if [[ -z "$FORCE" ]] ; then
     PREV_COMMIT=$(echo "${PREV}" | sed -e "s/-${BUILD_TYPE}.*$//")
   fi
 
+# Work out which candidate branch this one was probably forked from
+
+
   # If not found above, look for latest tagged
   if [[ -z ${PREV} ]] ; then
-    PREV=$(git log --oneline --no-merges | grep "^[0-9a-f]* Split off " | head -1 | sed "s/.*Split off //" | head -1)-rc1
+    best=1000000
+    for candidate in `git branch -r -l ${UPSTREAM}/candidate-8* | sed 's/\*//' | sed s/.*candidate-// | egrep -v [.]x | sort | uniq` ; do
+      delta=$(git rev-list --count --no-merges ${UPSTREAM}/candidate-$candidate...HEAD)
+      # echo $candidate : $delta
+      if [ $delta -lt $best ] ; then
+        best=$delta
+        PREV=`git tag -l | grep ${candidate} | tail -1 | sed s/.*_//`
```

Comment:
question: is it worth recording the best candidate in a var, and doing this outside of the for loop?

Reply:
Efficiency is very much second-place to clarity in this code... but the change you suggest MAY make it clearer.

Reply:
I decided I preferred it as it was